packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualLiquidatable.sol

```diff
@@ -252,11 +252,13 @@ contract PerpetualLiquidatable is PerpetualPositionManager {
             startCollateralNetOfWithdrawal = startCollateral.sub(positionToLiquidate.withdrawalRequestAmount);
         }
 
-        // Scoping to get rid of a stack too deep error.
+        // Scoping to get rid of a stack too deep error. In this block, we use the funding-rate adjusted value
+        // of token debt to check that the desired liquidation price is valid.
         {
-            FixedPoint.Unsigned memory startTokens = positionToLiquidate.tokensOutstanding;
+            FixedPoint.Unsigned memory startTokens =
```

Comment:
nit: shouldn't this be un-adjusted since it's compared to the user-provided values of collateral per token? And the user's idea of a token is non-adjusted?

Reply:
But `positionToLiquidate` is equal to `_getPositionData(sponsor)` so isnt it reading the sponsor's on-chain state?